A41-xds-rbac.md

```diff
@@ -0,0 +1,199 @@
+----
+* Author(s): [Eric Anderson](https://github.com/ejona86), [Ashitha
+  Santhosh](https://github.com/ashithasantosh) (RBAC engine behavior)
+* Approver: markdroth
+* Status: Draft {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2021-05-07
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Support the [xDS RBAC HTTP filter][RBAC filter] for service- and method-scoped
+client authorization (authz) on xDS-enabled gRPC servers.
+
+[RBAC filter]: https://github.com/envoyproxy/envoy/blob/main/api/envoy/extensions/filters/http/rbac/v3/rbac.proto
+
+## Background
+
+[A29 xDS-Based Security][A29] introduced the ability to manage xDS-managed mTLS
+meshes. As standard for TLS clients, the client verifies the server's identity
+is permitted to run the service. This is normally known as "hostname
+verification", but in a mesh scenario may be known as "server authorization" as
+the server's certificate does not include the hostname but other identity
+information.
+
+While servers authenticated clients (i.e., the client's identity was verified),
+A29 did not include mechanism to limit which clients were authorized to access a
+server and for which gRPC methods (i.e., whether the client was permitted).
+
+Envoy defines the [RBAC HTTP filter][RBAC filter] that is well-suited for
+limiting which nodes can access which services within a mesh as encouraged by
+the principal of least privilege.
+
+Resources in REST are present in the HTTP path so it would seem RBAC could be
+used in non-gRPC environments as a precise service-level resource authz engine.
+However, this 1) requires another filter to perform end-user authentication (as
+TLS just provides service-level authn) and 2) does not scale as RBAC provides no
+mechanism for loading rules on-demand. Since gRPC encodes service-level
+resource identifiers in the request message, RBAC does not have access to the
+requested resource for gRPC traffic.
+
+### Related Proposals:
+
+* [A36: xDS-Enabled Servers][A36]
+* [A29: xDS-Based Security for gRPC Clients and Servers][A29]
+* [A39: xDS HTTP Filter Support][A39]
+
+[A36]: https://github.com/grpc/proposal/blob/master/A36-xds-for-servers.md
+<!-- FIXME: replace with final link once merged -->
+[A29]: https://github.com/grpc/proposal/pull/184
+[A39]: https://github.com/grpc/proposal/blob/master/A39-xds-http-filters.md
+
+## Proposal
+
+Support the [xDS RBAC HTTP filter][RBAC filter] as a registered filter type on
+xDS-enabled gRPC Servers, and support its `RBACPerRoute` configuration override.
+This does _not_ include [RBAC network filter][] support nor running the filter
+on gRPC clients. xDS v2 support is not necessary.
+
+Supporting the RBAC HTTP filter on server-side leverages [A36: xDS-Enabled
+Servers][A36] and [A39: xDS HTTP Filter Support][A39]. Server-side HTTP filters
+are less common than client-side at this point in time, so this may be the first
```

Comment:
Instead of terms like "may be" and "likely", let's be concrete: We know that this is the first design to include supporting an HTTP filter on the server side, so this design will definitely need to describe how that will work.

A36 says "RouteConfiguration (via route_config or indirect fields like rds) is not used nor validated at this time, but may be in the future."  I think we need to update that here to describe exactly how the `RouteConfiguration` will be used.  (I'd also be fine with splitting that part into a separate gRFC if you'd prefer.)